Tasks/winget/main.ps1

```diff
@@ -155,37 +165,140 @@ function AppendToUserScript {
-    } else {
-        Invoke-CimMethod -ClassName Win32_Process -MethodName Create -Arguments @{CommandLine="C:\Program Files\PowerShell\7\pwsh.exe -MTA -Command `"Get-WinGetConfiguration -File $($ConfigurationFile) | Invoke-WinGetConfiguration -AcceptConfigurationAgreements`""}
+    }
+    else {
+        Write-Error "No package or configuration file specified"
+        exit 1
+    }
+}
+# We're running in the provisioning context:
+else {
+    Write-Host "Running in the provisioning context"
+
+    # We're running in package mode:
+    if ($Package) {
+        Write-Host "Running package install: $($Package)"
+        $processCreation = Invoke-CimMethod -ClassName Win32_Process -MethodName Create -Arguments @{CommandLine="C:\Program Files\PowerShell\7\pwsh.exe -MTA -Command `"Install-WinGetPackage -Id $($Package)`""}
+        $process = Get-Process -Id $processCreation.ProcessId
+        $handle = $process.Handle # cache process.Handle so ExitCode isn't null when we need it below
+        $process.WaitForExit()
+        $installExitCode = $process.ExitCode
+        if ($installExitCode -ne 0) {
+            Write-Error "Failed to install package. Exit code: $installExitCode"
+            exit 1
+        }
+    }
+    # We're running in configuration file mode:
+    elseif ($ConfigurationFile) {
+        if ($DownloadUrl) {
+            Write-Host "Downloading configuration file from: $($DownloadUrl)"
+            $ConfigurationFileDir = Split-Path -Path $ConfigurationFile
+            if(-Not (Test-Path -Path $ConfigurationFileDir))
+            {
+                New-Item -ItemType Directory -Path $ConfigurationFileDir
+            }
+    
+            Invoke-WebRequest -Uri $DownloadUrl -OutFile $ConfigurationFile
+        }
+
+        Write-Host "Running installation of configuration file: $($ConfigurationFile)"
+
+        $processCreation = Invoke-CimMethod -ClassName Win32_Process -MethodName Create -Arguments @{CommandLine="C:\Program Files\PowerShell\7\pwsh.exe -MTA -Command `"Get-WinGetConfiguration -File $($ConfigurationFile) | Invoke-WinGetConfiguration -AcceptConfigurationAgreements`""}
+        $process = Get-Process -Id $processCreation.ProcessId
+        $handle = $process.Handle # cache process.Handle so ExitCode isn't null when we need it below
+        $process.WaitForExit()
+        $installExitCode = $process.ExitCode
+        if ($installExitCode -ne 0) {
+            Write-Error "Failed to install packages. Exit code: $installExitCode"
+            exit 1
+        }
+    }
+    elseif ($InlineConfigurationBase64) {
+        Write-Host "Running installation of inline configuration"
+
+        $InlineConfiguration = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($InlineConfigurationBase64))
+
+        # if $ConfigurationFile is not specified, we need to write the configuration to a temporary file
+        if (-not $ConfigurationFile) {
```

Comment:
In this elseif block(ln 272), $Configuration would always be null or empty. So it woul be better to move the elseif block before ln 248.

Reply:
You're right. Good catch.